### DISCUSSION TITLE
How to change width of various fields of Form

### DISCUSSION THREAD

**Discussion by Manishkparmar80** [Budibase/budibase#1844] [Help] [answered]
I want to reduce width of Number field and Date field of form. how to do it ?

I tried to set field width under field design --> size --> width  but no effect.  
![set width](https://user-images.githubusercontent.com/84949590/123536554-8f7ec880-d748-11eb-9d55-3d78e56d45a1.JPG)

**Comment by Manishkparmar80** [comment, marked as answer]
@shogunpurple  can you pls. guide.

**Reply by shogunpurple** [reply]
@Manishkparmar80 please do not bump a thread that you raised just a few hours ago - we are answering all support queries as quickly as we can.

The easiest way to reduce the width of fields in a form is to reduce the width of the field group they are inside.

In order to reduce a single field, you will have to set the `display` property of that field to either `block` or `flex` in the design settings. This way, the field will respect the custom width you have set.

**Reply by Manishkparmar80** [reply]
@shogunpurple  I had set display property of field to flex and  then tried setting up width.  if i set width = 0 then it becoming too small and if i set size anything above 0 (e.g. 0.1) then it become full width.  it is not allowing to set in-between width. 
![set width1](https://user-images.githubusercontent.com/84949590/123550189-0f795280-d78a-11eb-9451-34fc7fbe18ae.jpg)
![set width2](https://user-images.githubusercontent.com/84949590/123550192-156f3380-d78a-11eb-9f7b-173b880408c8.jpg)

**Reply by shogunpurple** [reply]
Try setting a pixel width, like `50px`. I just tried this in the builder and it works as expected.

**Reply by Manishkparmar80** [reply]
yes it is working now as expected. thanks

**Reply by Manishkparmar80** [reply]
@shogunpurple i tried to resize attachment field.  Width and height is getting set as per pixel but still attachment field is not getting reduced equally. it is occupying bigger area of form, 

![attachement](https://user-images.githubusercontent.com/84949590/123551697-81549a80-d790-11eb-906b-188cde549a11.jpg)
![attachement2](https://user-images.githubusercontent.com/84949590/123551709-8b769900-d790-11eb-9fdc-103a1e208bd6.jpg)

**Reply by aptkingston** [reply]
@Manishkparmar80 you should put your fields inside a `FieldGroup` component - they are designed to work much better this way. Then in order to resize them you only need to change the width of the `FieldGroup` component itself.

![image](https://user-images.githubusercontent.com/9075550/123553451-23658a00-d773-11eb-8c63-77e3402902eb.png)
